### PR TITLE
Add clarity to readme

### PR DESCRIPTION
**Pull request by smothers**
Updates to the readme taking some of our most recent conversations into account. I tried to disambiguate the API so that all the features are documented if only briefly. This was written on my iPad and I don’t think spellcheck was working so need to give it another pass for spelling.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,38 +24,51 @@ You can clone and run our React code examples [here](https://github.com/homebase
...
+  // `unique: 'identity` which enforces a uniqueness constraint 
+  // and let's you lookup entites by their unique attributes.
   schema: {
     todo: {
       project: { type: 'ref' },
       name: { unique: 'identity' }
     }
   },
+  
+  // Initial data is like it sound.
```
Comment (rubinovitz) [inline README.md:65]: *sounds

**Comment by rubinovitz** [comment]
@smothers I feel like this is so much more true that what we want to do and speaks to the user feedback we've gotten really well. I love it! Added a typo correction but otherwise looks great to me. Excited to keep dialing this in :)

File: README.md (hunk 1 above)
Comment (smothers) [reply README.md:65]: ty

**Review by rubinovitz — APPROVED** [review]
LGTM 🚀

**Pull request merged**